src/zenml/logging/step_logging.py

```diff
@@ -78,13 +89,118 @@ def prepare_logs_uri(
         artifact_store.makedirs(logs_base_uri)
 
     # Delete the file if it already exists
-    logs_uri = os.path.join(logs_base_uri, f"{log_key}.log")
-    if artifact_store.exists(logs_uri):
-        logger.warning(
-            f"Logs file {logs_uri} already exists! Removing old log file..."
+    if artifact_store.config.IS_IMMUTABLE_FILESYSTEM:
+        logs_uri_folder = os.path.join(logs_base_uri, log_key)
+        if artifact_store.exists(logs_uri_folder):
+            logger.warning(
+                f"Logs directory {logs_uri_folder} already exists! Removing old log directory..."
+            )
+            artifact_store.rmtree(logs_uri_folder)
+
+        artifact_store.makedirs(logs_uri_folder)
+        return logs_uri_folder
+    else:
+        logs_uri = os.path.join(logs_base_uri, f"{log_key}.log")
+        if artifact_store.exists(logs_uri):
+            logger.warning(
+                f"Logs file {logs_uri} already exists! Removing old log file..."
+            )
+            artifact_store.remove(logs_uri)
+        return logs_uri
+
+
+def fetch_logs(
+    zen_store: "BaseZenStore",
+    artifact_store_id: Union[str, UUID],
+    logs_uri: str,
+    offset: int = 0,
+    length: int = 1024 * 1024 * 16,  # Default to 16MiB of data
+) -> str:
+    """Fetches the logs from the artifact store.
+
+    Args:
+        zen_store: The store in which the artifact is stored.
+        artifact_store_id: The ID of the artifact store.
+        logs_uri: The URI of the artifact.
+        offset: The offset from which to start reading.
+        length: The amount of bytes that should be read.
+
+    Returns:
+        The logs as a string.
+
+    Raises:
+        DoesNotExistException: If the artifact does not exist in the artifact
+            store.
+    """
+
+    def _read_file(
+        uri: str, offset: int = 0, length: Optional[int] = None
+    ) -> str:
+        return str(
+            _load_file_from_artifact_store(
+                uri,
+                artifact_store=artifact_store,
+                mode="rb",
+                offset=offset,
+                length=length,
+            ).decode()
         )
-        artifact_store.remove(logs_uri)
-    return logs_uri
+
+    artifact_store = _load_artifact_store(artifact_store_id, zen_store)
+    if logs_uri.endswith(LOGS_EXTENSION):
```

Comment:
Given that we now have a more reliable way to differentiate between immutable and mutable, could this be:

```suggestion
    if artifact_store.config.IS_IMMUTABLE_FILESYSTEM:
```

Reply:
Agree, but I'd better check for `not artifact_store.isdir(logs_uri)`, so it is more generic. A file might be already on Immutable FS, but not be a folder. This is valid for legacy objects.